### ISSUE TITLE
[BUG] Dataclass `'NoneType' object is not iterable`

### ISSUE THREAD

**Issue by AlecHaring** [dashingsoft/pyarmor#1067]
BCC mode throws the following error when a dataclass field has the type `None`, or `Union[..., None]`:
```
'NoneType' object is not iterable
```

**Executed command:** `pyarmor -d gen --enable-bcc .\pyarmor_test\dataclass_test.py`

**dataclass_test.py:**
```python
from dataclasses import dataclass
from typing import Union


@dataclass
class TestData:
    test_field: Union[str, None]

```

**pyarmor.error.log**
```
11800 MainProcess 2023-04-01 14:33:40,213
Traceback (most recent call last):
  File "C:\Users\alec\Desktop\PythonProjects\pyarmor-test\venv\lib\site-packages\pyarmor\cli\__main__.py", line 627, in main
    main_entry(sys.argv[1:])
  File "C:\Users\alec\Desktop\PythonProjects\pyarmor-test\venv\lib\site-packages\pyarmor\cli\__main__.py", line 615, in main_entry
    args.func(ctx, args)
  File "C:\Users\alec\Desktop\PythonProjects\pyarmor-test\venv\lib\site-packages\pyarmor\cli\__main__.py", line 179, in cmd_gen
    builder.process(options, pack=args.pack)
  File "C:\Users\alec\Desktop\PythonProjects\pyarmor-test\venv\lib\site-packages\pyarmor\cli\generate.py", line 153, in process
    self._obfuscate_scripts()
  File "C:\Users\alec\Desktop\PythonProjects\pyarmor-test\venv\lib\site-packages\pyarmor\cli\generate.py", line 118, in _obfuscate_scripts
    code = Pytransform3.generate_obfuscated_script(self.ctx, r)
  File "C:\Users\alec\Desktop\PythonProjects\pyarmor-test\venv\lib\site-packages\pyarmor\cli\core\__init__.py", line 48, in generate_obfuscated_script
    return m.generate_obfuscated_script(ctx, res)
  File "<maker>", line 719, in generate_obfuscated_script
  File "<maker>", line 509, in process
  File "<maker>", line 509, in <listcomp>
  File "<bccmaker>", line 860, in process
TypeError: 'NoneType' object is not iterable
```

**Comment by jondy** [comment]
Got it. I'll check it

**Comment by jondy** [comment]
Reproduced, it will be fixed in next dev release

**Comment by jondy** [comment]
Fixed in dev version `8.1.dev2`
